hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -2261,6 +2259,44 @@ public boolean isDefinedOnSubclass(int i) {
 		return Arrays.copyOf( fields, counter );
 	}
 
+	@Override
+	public int[] resolveDirtyAttributeIndexes(String[] attributeNames) {
+		final int[] mutablePropertiesIndexes = entityMetamodel.getMutablePropertiesIndexes();
```

Comment:
This looks a bit scary, but after reading through it it seems all it does is a union Set of the indexes of mutable properties + the other named attributes.
Did I read that correctly? Maybe we could add an implementation comment?

Reply:
I will adapt the implementation a bit as it is not 100% correct yet. Working on it atm.